packages/containers/src/List/utils.js

```diff
@@ -0,0 +1,101 @@
+import cmf from '@talend/react-cmf';
+import get from 'lodash/get';
+import { Map, List } from 'immutable';
+
+function contains(listItem, query, columns) {
+	let item = listItem;
+	if (Map.isMap(listItem)) {
+		item = listItem.toJS();
+	}
+	for (const column of columns) {
+		if (typeof item[column.key] === 'string') {
+			if (item[column.key].toLowerCase().indexOf(query.toLowerCase()) !== -1) {
+				return true;
+			}
+		}
+	}
+	return false;
+}
+
+function getCollection(state, collectionId) {
+	return state.cmf.collections.get(collectionId);
+}
+
+export function getCollectionItems(state, collectionId) {
+	const collection = getCollection(state, collectionId);
+
+	if (Map.isMap(collection)) {
+		return collection.get('items');
+	}
+	return collection;
+}
+
+export function configureGetPagination(state, { collectionId }) {
+	const collection = getCollection(state, collectionId);
+
+	if (Map.isMap(collection)) {
+		return collection.get('pagination');
+	}
+
+	return null;
+}
+
+export function getComponentState(collectionId) {
+	return state => state.cmf.components.getIn(['Container(List)', collectionId || 'default']);
+}
+
+export function sortList(componentState, localConfig, items) {
+	if (!List.isList(items)) {
+		return new List();
+	}
+	let results = items;
+	if (componentState) {
+		const sortBy = componentState.get('sortOn');
+		const sortAsc = componentState.get('sortAsc');
+		let compare = (a, b) => {
+			if (a.get(sortBy)) {
+				if (a.get(sortBy).localCompare) {
+					return a.get(sortBy).localeCompare(b.get(sortBy));
+				}
+				const aValue = `${a.get(sortBy) || ''}`.toLowerCase();
+				const bValue = `${b.get(sortBy) || ''}`.toLowerCase();
+				if (aValue < bValue) {
+					return -1;
+				}
+				if (aValue > bValue) {
+					return 1;
+				}
+				return 0;
+			}
+			if (!b[sortBy]) {
+				return 0;
+			}
+			return -1;
+		};
+
+		const sortedColumn = get(localConfig, 'columns', [])
+			.filter(column => column.key === sortBy)
+			.pop();
+		if (sortedColumn && sortedColumn.sortFunction) {
+			compare = cmf.registry.getFromRegistry(sortedColumn.sortFunction)(sortBy);
+		}
+		results = results.sort(compare);
+		if (!sortAsc) {
+			results = results.reverse();
```

Comment:
the `asc` or `desc` should be done directly in the sort method.
You can use currying for creating the compare fn like this :

```javascript
const compareBy = sortAsc => (aIn, bIn) => {
   // you can inverse 'a' and 'b' if in descending with some other variables
   const a = sortAsc ? aIn : bIn;
   const b = sortAsc ? bIn : aIn;
   ...
}

------

results.sort(compareBy(sortAsc));
```

Reply:
After that the ascending check is not more needed in the top level of the function. We can now just return the result of the sorting. And BTW, we can directly use the `items` parameter without assign it to an other variable so :

```javascript
return items.sort(compareBy(sortAsc));
```